src/matlab2tikz.m

```diff
@@ -1911,7 +1915,23 @@ function assertRegularAxes(handle)
     matlabDefaultLineWidth = 0.5;
     if m2t.args.strict ...
             || ~abs(lineWidth-matlabDefaultLineWidth) <= m2t.tol
-        lineOpts = opts_add(lineOpts, 'line width', sprintf('%.1fpt', lineWidth));
+        if m2t.args.semanticalLineWidth
+            semStrID = find(lineWidth == m2t.args.semanticalValueList);
+            if isempty(semStrID)
+                lineOpts = opts_add(lineOpts, 'line width', sprintf('%.1fpt', lineWidth));
+            else
+                lineOpts = opts_add(lineOpts, m2t.args.semanticalStringList{semStrID}, []);
```

Comment:
I think the empty array at the end is superfluous as we default to empty values anyway